applications/crossbar/src/modules/phonebook.erl

```diff
@@ -0,0 +1,195 @@
+%%%-----------------------------------------------------------------------------
+%%% @copyright (C) 2012-2018, 2600Hz
+%%% @doc Common functions for interacting with the 2600Hz phonebook service
+%%% @author Sean Wysor
+%%% @end
+%%%-----------------------------------------------------------------------------
+-module(phonebook).
+
+-export([maybe_create_port_in/1
+        ,maybe_add_comment/2
+        ,maybe_cancel_port_in/1
+        ]).
+
+-include("crossbar.hrl").
+
+-define(MOD_CONFIG_CAT, <<"crossbar.phonebook">>).
+
+
+%%------------------------------------------------------------------------------
+%% @doc exported functions
+%% @end
+%%------------------------------------------------------------------------------
+
+-spec maybe_create_port_in(cb_context:context()) -> cb_context:context().
+maybe_create_port_in(Context) ->
+    case send_to_phonebook(Context) of
+        'true' ->
+            create_port_in(cb_context:doc(Context), cb_context:auth_token(Context)),
+            Context;
+        'false' -> Context
+    end.
+
+-spec maybe_add_comment(cb_context:context(), kz_term:ne_binary()) -> cb_context:context().
+maybe_add_comment(Context, Comment) ->
+    case send_to_phonebook(Context)
+        andalso not req_from_phonebook(Context)
+    of
+        'true' ->
+            add_comment(cb_context:doc(Context), cb_context:auth_token(Context), Comment),
+            Context;
+        'false' -> Context
+    end.
+
+-spec maybe_cancel_port_in(cb_context:context()) -> cb_context:context().
+maybe_cancel_port_in(Context) ->
+    case send_to_phonebook(Context) of
+        'true' ->
+            cancel_port_in(cb_context:doc(Context), cb_context:auth_token(Context)),
+            Context;
+        'false' -> Context
+    end.
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% @end
+%%------------------------------------------------------------------------------
+
+-spec phonebook_enabled() -> boolean().
+phonebook_enabled() ->
+    kapps_config:get_is_true(?MOD_CONFIG_CAT, <<"enabled">>, 'false').
+
+-spec req_from_phonebook(cb_context:context()) -> boolean().
+req_from_phonebook(Context) ->
+    case cb_context:req_header(Context, <<"User-Agent">>) of
+        <<"phonebook">> -> 'true';
+        _ -> 'false'
+    end.
+
+send_to_phonebook(Context) ->
+    cb_context:resp_status(Context) =:= 'success'
+        andalso phonebook_enabled().
+
+-spec create_port_in(kz_json:object(), kz_term:ne_binary()) -> 'ok'.
+create_port_in(JObj, AuthToken) ->
+    send_req('create_port_in'
```

Comment:
So you don't really use this atom for much in `send_req/5`. Meaning `send_req` isn't really a generic function since each clause just matches the atom in the first arg to do its logic.

Instead, I would use these named functions to create the specific pieces (URL, req body/headers) then call a generic make_request with the necessary bits. Does that make sense?